### PR TITLE
Feat/water 2746 TPT integration test (FORK)

### PR DESCRIPTION
**Pull request by timtamimi**
This PR changes the way integration tests are run. 

To run the full suite, please use:

`npm run test:integration-sequential`

To run an individual test in isolation, please use:

`npm run test:integration integration-tests/path/to/test.js`

### CODE REVIEW

**Review by james-siteclick — COMMENTED** [review]
I have made changes to this:
* Fixed issues with authentication of routes requiring an IDM user that did not exist
* Fixed data/relationship issues with CRM fixtures 
* Added new code to load fixtures from YAML files
* Tests now run concurrently without issue, so removed sequential test mentioned above

File: integration-tests/billing/services/licence-agreements.js [hunk 1]
```diff
@@ -1,36 +1,10 @@
...
-};
+const { bookshelf } = require('../../../src/lib/connectors/bookshelf');
 
 const tearDown = () =>
   bookshelf.knex('water.licence_agreements')
     .where('is_test', true)
     .del();
 
-exports.create = create;
+// exports.create = create;
```
Comment (DGajewska) [inline integration-tests/billing/services/licence-agreements.js:9]: Did you mean to leave this commented out?

File: integration-tests/billing/services/fixture-loader/FixtureLoader.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    for (const config of data) {
+      // Pre-process field data to include references to previously inserted models
+      // Models are processed sequencially
+      const data = mapValues(config.fields, val => mapValue(val, _refs));
+
+      // Create new model using adapter
+      const model = await this._adapter.create(config, data);
+
+      console.log(model);
```
Comment (DGajewska) [inline integration-tests/billing/services/fixture-loader/FixtureLoader.js:89]: Is this console.log deliberate?

File: integration-tests/billing/tests/supplementary/SB2.js [hunk 3]
```diff
@@ -124,43 +120,24 @@ experiment('supplementary ref: SB2', () => {
...
-          const transaction = licence.billingTransactions.find(tx => tx.isCredit === true);
+          const transaction = licence.billingTransactions.find(tx => tx.isCredit === true && tx.chargeType === 'standard');
           expect(transaction).to.exist();
         });
 
         experiment('the first debit transaction', () => {
           let transaction;
           beforeEach(async () => {
-            transaction = licence.billingTransactions.find(tx => tx.isCredit === false && tx.description === 'CE3');
+            transaction = licence.billingTransactions.find(tx => tx.transactionKey === '1e1fee0c4146c314e12f5f9067b25b24');
```
Comment (DGajewska) [inline integration-tests/billing/tests/supplementary/SB2.js:140]: Doesn't this make the 'has a stable transaction key' test redundant?

**Review by DGajewska — APPROVED** [review]

File: integration-tests/billing/services/licence-agreements.js (hunk 1 above)
Comment (james-siteclick) [reply integration-tests/billing/services/licence-agreements.js]: Fixed

File: integration-tests/billing/services/fixture-loader/FixtureLoader.js (hunk 2 above)
Comment (james-siteclick) [reply integration-tests/billing/services/fixture-loader/FixtureLoader.js]: Fixed

File: integration-tests/billing/tests/supplementary/SB2.js (hunk 3 above)
Comment (james-siteclick) [reply integration-tests/billing/tests/supplementary/SB2.js]: Fixed

**Pull request merged**